Lua/Lua.sublime-syntax

```diff
@@ -1,87 +1,480 @@
+      scope: punctuation.section.group.begin.lua
+      set:
+        - meta_scope: meta.group.lua
+        - match: \)
+          scope: punctuation.section.group.end.lua
+          pop: true
+        - match: (?=\S)
+          push: expression-list
+
+  prefix-operator:
+    - match: '#'
+      scope: keyword.operator.length.lua
+    - match: \-
+      scope: keyword.operator.arithmetic.lua
+    - match: \~
+      scope: keyword.operator.bitwise.lua
+    - match: not{{identifier_break}}
+      scope: keyword.operator.logical.lua
+
+  builtin:
+    - match: true{{identifier_break}}
+      scope: constant.language.boolean.true.lua
+      pop: true
+    - match: false{{identifier_break}}
+      scope: constant.language.boolean.true.lua
+      pop: true
+    - match: nil{{identifier_break}}
+      scope: constant.language.null.lua
+      pop: true
+    - match: \.\.\.
       scope: constant.language.lua
-    - match: '(?<![^.]\.|:)\b(self)\b'
-      scope: variable.language.self.lua
-    - match: '(?<![^.]\.|:)\b(assert|collectgarbage|dofile|error|getfenv|getmetatable|ipairs|loadfile|loadstring|module|next|pairs|pcall|print|rawequal|rawget|rawset|require|select|setfenv|setmetatable|tonumber|tostring|type|unpack|xpcall)\b(?=[( {"''\[])'
-      scope: support.function.lua
-    - match: '(?<![^.]\.|:)\b(coroutine\.(create|resume|running|status|wrap|yield)|string\.(byte|char|dump|find|format|gmatch|gsub|len|lower|match|rep|reverse|sub|upper)|table\.(concat|insert|maxn|remove|sort)|math\.(abs|acos|asin|atan2?|ceil|cosh?|deg|exp|floor|fmod|frexp|ldexp|log|log10|max|min|modf|pow|rad|random|randomseed|sinh?|sqrt|tanh?)|io\.(close|flush|input|lines|open|output|popen|read|tmpfile|type|write)|os\.(clock|date|difftime|execute|exit|getenv|remove|rename|setlocale|time|tmpname)|package\.(cpath|loaded|loadlib|path|preload|seeall)|debug\.(debug|[gs]etfenv|[gs]ethook|getinfo|[gs]etlocal|[gs]etmetatable|getregistry|[gs]etupvalue|traceback))\b(?=[( {"''\[])'
-      scope: support.function.library.lua
-    - match: \b(and|or|not)\b
-      scope: keyword.operator.lua
-    - match: '\+|-|%|#|\*|\/|\^|==?|~=|<=?|>=?|(?<!\.)\.{2}(?!\.)'
-      scope: keyword.operator.lua
+      pop: true
+    - match: self{{identifier_break}}
+      scope: variable.language.this.lua
+      pop: true
+
+  variable:
+    - match: '{{identifier}}{{function_assignment_ahead}}'
+      scope: entity.name.function.lua
+      pop: true
+
+    - match: '{{identifier}}{{function_call_ahead}}'
+      scope: variable.function.lua
+      pop: true
+
+    - match: '{{identifier}}'
+      scope: variable.other.lua
+      pop: true
+
+  number:
+    - match: 0[Xx]\h*(?:\.\h*)?([Pp][-+]?\d*)?
+      scope: constant.numeric.hexadecimal.lua
+      pop: true
+
+    - match: \d+(?:\.\d*)?([Ee][-+]?\d*)?
+      scope: constant.numeric.decimal.lua
+      pop: true
+
+  string:
+    - match: \'
+      scope: punctuation.definition.string.begin.lua
+      set:
+        - meta_scope: string.quoted.single.lua
```

Comment:
The prototype context should probably not be included:

```
local foo = "bar--baz"
print(foo) -- prints "bar--baz"
```

Reply:
Fixed.